### ISSUE TITLE
Better support for surface layer timeseries

### ISSUE THREAD

**Issue by sofroniewn** [napari/napari#829]
## 🚀 Feature
As discussed in this [image.sc post](https://forum.image.sc/t/slider-widget-to-control-surface-values/32482/6) it would be nice to provide better support for people with timeseries values of static surfaces. It would be quite easy to support an API where people could pass data in the form
```
vertices N x 3 (constant: doesn’t change with time)
faces M x 3 (constant: doesn’t change with time)
values L x N (each time has N values for the vertices )
```
and we could take care of the rest. Note that vertices could still be `N x D`, and values could now be `D'` dimensional and we'd just convert all the `D'-1` dimensions into additional slider like dimensions. Should be quite clean and conform to users expectations much more strongly.

I can work on this when I get the chance, shouldn't be too bad

**Comment by jni** [comment]
@sofroniewn it's unclear right now whether this is exactly the right data model. I'd like us to look at datasets conforming to this format, e.g. fMRI datasets, and see how the data usually comes, and then try to tailor our API to that.

**Comment by sofroniewn** [comment]
Working through an example from [nilearn](https://nilearn.github.io/auto_examples/01_plotting/plot_surf_stat_map.html) with surface timeseries data I encountered the following:

```python
# NKI resting state data from nilearn
from nilearn import datasets
from nilearn import surface

nki_dataset = datasets.fetch_surf_nki_enhanced(n_subjects=1)

# Fsaverage5 surface template
fsaverage = datasets.fetch_surf_fsaverage()

# Load surface data and resting state time series from nilearn
pial_left = surface.load_surf_data(fsaverage['pial_left']) # 2-tuple
# pial_left[0].shape (10242, 3) (vertices)
# pial_left[1].shape (20480, 3) (faces)
sulc_left = surface.load_surf_data(fsaverage['sulc_left']) # shape (10242,)
timeseries = surface.load_surf_data(nki_dataset['func_left'][0]) # shape (10242, 895)
``` 
Then viewing in napari
```python
import napari
viewer = napari.Viewer(ndisplay=3)
viewer.add_surface((pial_left[0], pial_left[1], sulc_left))
viewer.add_surface((pial_left[0], pial_left[1], timeseries[:, 0]), colormap='green') # just one timeslice
```

Right now creating and viewing the "timeseries compatible representation" can be done with
```python
import numpy as np
vertices = pial_left[0]
faces = pial_left[1]
values = timeseries.T

n = len(vertices)
k = len(values)
vertices_t = np.concatenate(
    [np.concatenate((np.full((n, 1), i), vertices), axis=1) for i in range(k)],
    axis=0,
)
faces_t = np.concatenate(
    [faces + n*i for i in range(k)],
    axis=0,
)
values_t = np.concatenate([values[i] for i in range(k)])
viewer.add_surface((vertices_t, faces_t, values_t), colormap='red')
``` 

But the data is so large that viewing performance is now terrible, the refresh is too slow to be useable (though it does look cool - see below).

I think my
```
vertices N x D
faces M x 3
values L1 x L2 x LK x N
```
representation, which will result in an `D+K` dimensional layer isn't too far off. We can decide though if we want
```python
data = (vertices, faces)
values = values
```
or
```python
data = (vertices, faces, values)
```
to be the main API. We should also do something sensible if values is not provided. Right now, it's a bit awkward as you always have to give them in the 3-tuple. I maybe prefer the `data` and `values` separate API, it might also make this new functionality easier to understand, or make it easier to provide values for faces one day and not just vertices.

Here is the brain surface with the slider:
![image](https://user-images.githubusercontent.com/6531703/71450489-f9fc1600-2717-11ea-88df-132927d0106e.png)

In general [nilearn plotting](https://nilearn.github.io/modules/reference.html#module-nilearn.plotting) is a good place to look in this context, and I am sure we could reach out to them for questions/ advice

**Comment by jni** [comment]
@sofroniewn brilliant, this is a great example from which to get started, and indeed it seems like the API you originally suggested is well-matched to the nilearn example data. We do need to think about cases when the surface itself might be evolving though, and how to disambiguate such things.

We'll indeed have to think about the API, and whether to split out the values. Also, I feel like the `values` parameter should be renamed to `vertex_values`, though I don't know whether I'm rehashing old arguments here. But it would make it easier for newcomers to get their head around data, if they are unfamiliar with it.

**Comment by sofroniewn** [comment]
I'm open to `data` a 2-tuple of (vertices, faces), and `vertex_values` an array. I think this makes it clear that the `data` and `vertex_values` are separate, and if you have a 3D evolving surface you should have 4D `data` and 1D `vertex_values`, but if you have a 3D static surface you can just do 3D `data` and 2D `vertex_values`.

We can wait on that one until after we see the implementation as I think it should be pretty easy to change

**Comment by sofroniewn** [comment]
I did not realize this is a duplicate of #540
